addons/hr_holidays/models/hr_leave.py

```diff
@@ -119,6 +136,17 @@ def _compute_can_reset(self):
             if group_hr_manager in user.groups_id or holiday.employee_id and holiday.employee_id.user_id == user:
                 holiday.can_reset = True
 
+    @api.depends('employee_id', 'department_id')
+    def _compute_can_approve(self):
+        """ User can not approve a leave request if is its own leave request
+            except if he is hr_manager and he has no manager
+        """
+        for holiday in self:
+            manager = self.user_has_groups('hr_holidays.group_hr_holidays_manager') \
```

Comment:
I don't understand that line. I think it does not match class docstring and/or method docstring. Moreover understand a triple and not + an or is not really easy.